pom.xml

```diff
@@ -4,482 +4,662 @@
 
-<project xmlns="http://maven.apache.org/POM/4.0.0"
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-    <modelVersion>4.0.0</modelVersion>
+  <groupId>com.snowflake</groupId>
+  <artifactId>snowflake-kafka-connector</artifactId>
+  <version>1.9.1</version>
+  <packaging>jar</packaging>
 
-    <groupId>com.snowflake</groupId>
-    <artifactId>snowflake-kafka-connector</artifactId>
-    <version>1.9.1</version>
-    <packaging>jar</packaging>
-    <name>Snowflake Kafka Connector</name>
-    <description>Snowflake Kafka Connect Sink Connector</description>
-    <url>https://www.snowflake.com/</url>
+  <name>Snowflake Kafka Connector</name>
+  <description>Snowflake Kafka Connect Sink Connector</description>
+  <url>https://www.snowflake.com/</url>
+  <licenses>
+    <license>
+      <name>Apache License, Version 2.0</name>
+      <url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
+      <distribution>repo</distribution>
+    </license>
+  </licenses>
 
-    <developers>
-        <developer>
-            <name>Snowflake Support Team</name>
-            <email>snowflake-java@snowflake.com</email>
-            <organization>Snowflake Computing</organization>
-            <organizationUrl>https://www.snowflake.com</organizationUrl>
-        </developer>
-    </developers>
+  <developers>
+    <developer>
+      <name>Snowflake Support Team</name>
+      <email>snowflake-java@snowflake.com</email>
+      <organization>Snowflake Computing</organization>
+      <organizationUrl>https://www.snowflake.com</organizationUrl>
+    </developer>
+  </developers>
 
-    <licenses>
-        <license>
-            <name>Apache License, Version 2.0</name>
-            <url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
-            <distribution>repo</distribution>
-        </license>
-    </licenses>
+  <scm>
+    <connection>scm:git:https://github.com/snowflakedb/snowflake-kafka-connector.git</connection>
+    <url>https://github.com/snowflakedb/snowflake-kafka-connector</url>
+  </scm>
+  <distributionManagement>
+    <snapshotRepository>
+      <id>ossrh</id>
+      <url>https://oss.sonatype.org/content/repositories/snapshots</url>
+    </snapshotRepository>
+  </distributionManagement>
 
-    <scm>
-        <connection>
-            scm:git:https://github.com/snowflakedb/snowflake-kafka-connector.git
-        </connection>
-        <url>https://github.com/snowflakedb/snowflake-kafka-connector</url>
-    </scm>
+  <properties>
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <maven.compiler.target>1.8</maven.compiler.target>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <skipIntegrationTests>${skipTests}</skipIntegrationTests>
+    <skipTests>false</skipTests>
+    <skipUnitTests>${skipTests}</skipUnitTests>
+    <version.dependency.avro>1.11.1</version.dependency.avro>
```

Comment:
This looks great, thanks a lot for opening this PR and adding bunch of great/recommended things.